ros2topic/ros2topic/verb/hz.py

```diff
@@ -0,0 +1,339 @@
+        return
+
+
+def _rostopic_hz(node, topic, window_size=DEFAULT_WINDOW_SIZE, filter_expr=None, use_wtime=False):
+    """
+    Periodically print the publishing rate of a topic to console until shutdown.
+
+    :param topic: topic name, ``list`` of ``str``
+    :param window_size: number of messages to average over, -1 for infinite, ``int``
+    :param filter_expr: Python filter expression that is called with m, the message instance
+    """
+    # pause hz until topic is published
+    msg_class = get_msg_class(node, topic, blocking=True)
+
+    if msg_class is None:
+        node.destroy_node()
+        return
+
+    rt = ROSTopicHz(node, window_size, filter_expr=filter_expr, use_wtime=use_wtime)
+    node.create_subscription(
+        msg_class,
+        topic,
+        functools.partial(rt.callback_hz, topic=topic),
+        qos_profile=qos_profile_sensor_data)
+
+    while rclpy.ok():
+        rclpy.spin_once(node)
+        rt.print_hz(topic)
+
+    node.destroy_node()
+    rclpy.shutdown()
+
+
+def get_msg_class(node, topic, blocking=False):
+    msg_class = _get_msg_class(node, topic)
+    if msg_class:
+        return msg_class
+    elif blocking:
+        print('WARNING: topic [%s] does not appear to be published yet' % topic)
+        while rclpy.ok():
+            msg_class = _get_msg_class(node, topic)
+            if msg_class:
+                return msg_class
+            else:
+                sleep(0.1)
+    else:
+        print('WARNING: topic [%s] does not appear to be published yet' % topic)
+    return None
+
+
+def _get_msg_class(node, topic):
+    """
+    Get message module based on topic name.
+
+    :param topic: topic name, ``list`` of ``str``
+    """
+    topic_names_and_types = get_topic_names_and_types(node=node)
+    try:
+        expanded_name = expand_topic_name(topic, node.get_name(), node.get_namespace())
+    except ValueError as e:
+        raise RuntimeError(e)
+    try:
+        validate_full_topic_name(expanded_name)
+    except rclpy.exceptions.InvalidTopicNameException as e:
+        raise RuntimeError(e)
+    for n, t in topic_names_and_types:
+        if n == expanded_name:
+            if len(t) > 1:
+                raise RuntimeError(
+                    "Cannot echo topic '%s', as it contains more than one type: [%s]" %
+                    (topic, ', '.join(t))
+                )
+            message_type = t[0]
+            break
+    else:
+        # Could not determine the type for the passed topic
+        return None
+
+    # TODO(dirk-thomas) this logic should come from a rosidl related package
+    try:
+        package_name, message_name = message_type.split('/', 2)
+        if not package_name or not message_name:
+            raise ValueError()
+    except ValueError:
+        raise RuntimeError('The passed message type is invalid')
+    module = importlib.import_module(package_name + '.msg')
+    return getattr(module, message_name)
```

Comment:
The two methods `get_msg_class` and `_get_msg_class` are a 1-to-1 copy from the `delay` module. Please don't cop-n-paste the code but move it into the `api` module and reuse it in both locations.